Mage.Sets/src/mage/cards/t/TheEnd.java

```diff
@@ -0,0 +1,95 @@
+import mage.abilities.effects.common.search.SearchTargetGraveyardHandLibraryForCardNameAndExileEffect;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.CardType;
+import mage.constants.Zone;
+import mage.filter.FilterCard;
+import mage.filter.predicate.mageobject.NamePredicate;
+import mage.game.Game;
+import mage.game.permanent.Permanent;
+import mage.players.Player;
+import mage.target.common.TargetCreatureOrPlaneswalker;
+
+import java.util.UUID;
+
+/**
+ *
+ * @author Susucr
+ */
+public final class TheEnd extends CardImpl {
+
+    public TheEnd(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.INSTANT}, "{2}{B}{B}");
+
+        // This spell costs {2} less to cast if your life total is 5 or less.
+        this.addAbility(new SimpleStaticAbility(
+                Zone.ALL,
+                new SpellCostReductionSourceEffect(
+                        2,
+                        new XorLessLifeCondition(XorLessLifeCondition.CheckType.CONTROLLER, 5)
+                ).setCanWorksOnStackOnly(true).setText("This spell costs {2} less to cast if your life total is 5 or less.")
+        ).setRuleAtTheTop(true));
+
+        // Exile target creature or planeswalker. Search its controller's graveyard, hand, and library for any number of cards with the same name as that permanent and exile them. That player shuffles, then draws card for each card exiled from their hand this way.
+        this.getSpellAbility().addEffect(new TheEndEffect());
+        this.getSpellAbility().addTarget(new TargetCreatureOrPlaneswalker());
+    }
+
+    private TheEnd(final TheEnd card) {
+        super(card);
+    }
+
+    @Override
+    public TheEnd copy() {
+        return new TheEnd(this);
+    }
+}
+
+class TheEndEffect extends SearchTargetGraveyardHandLibraryForCardNameAndExileEffect {
+
+    TheEndEffect() {
+        super(true, "its controller's", "any number of cards with the same name as that permanent");
+        this.staticText = "Exile target creature or planeswalker. Search its controller's graveyard, hand, and library for any number of cards with the same name as that permanent and exile them. That player shuffles, then draws card for each card exiled from their hand this way";
+    }
+
+    private TheEndEffect(final TheEndEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public TheEndEffect copy() {
+        return new TheEndEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Permanent permanent = game.getPermanent(source.getFirstTarget());
+        if(permanent == null) {
+            return false;
+        }
+        String name = permanent.getName();
+        Player player = game.getPlayer(permanent.getControllerId());
+        if(player == null) {
+            return false;
+        }
+
+        permanent.moveToExile(null, "", source, game);
```

Comment:
This seems like an inner method, not to be called from a card.

Normally you'd use player.moveCards or something like.

Reply:
Sure I'll change that.
Is in used in many cards code though
![image](https://github.com/magefree/mage/assets/34709007/a60fa8ec-79db-49cc-912c-c428a62e8f3a)